metric/registry.go

```diff
@@ -15,15 +15,18 @@
 package metric
 
 import (
-	"go.opencensus.io/metric/metricdata"
 	"log"
+	"sync"
 	"time"
+
+	"go.opencensus.io/metric/metricdata"
 )
 
 // Registry creates and manages a set of gauges.
 // External synchronization is required if you want to add gauges to the same
 // registry from multiple goroutines.
 type Registry struct {
+	mu     sync.RWMutex
 	gauges map[string]*gauge
```

Comment:
Maybe consider to use sync.Map? Please let me know if that matches your requirements.

Reply:
sync.Map would work.